### PR TITLE
Remove references to the develop branch

### PR DESCRIPTION
**Pull request by LucioFranco**
This change is related to amethyst/amethyst#924.

This is related to #101

### CODE REVIEW

File: generate.sh [hunk 1]
```diff
@@ -33,33 +32,74 @@ cargo install-update --version || cargo install cargo-update;
 mdbook --version || cargo install mdbook;
 cargo install-update -a;
 
-for (( i=0; i<${#branches[@]}; i++ ));
+git clone https://github.com/amethyst/amethyst --branch master amethyst/master
+cd amethyst/master
+
+cargo doc --all --no-deps;
```
Comment (AnneKitsune) [inline generate.sh]: is this why we don't have access to structs of our dependencies in the docs?

File: generate.sh [hunk 2]
```diff
@@ -33,33 +32,74 @@ cargo install-update --version || cargo install cargo-update;
...
+cp -r amethyst/master/target/doc/ build/doc/latest/
+
+mkdir -p build/book/latest/
+mv -f amethyst/master/book/book/* build/book/latest/
+
+touch build/doc/latest/VERSION
+echo "$LATEST_TAG" > build/doc/latest/VERSION
+
+cd amethyst/master
+for tag in $(git tag)
```
Comment (AnneKitsune) [inline generate.sh:82]: Why are we building a book for each tag? Isn't that the job of docs.rs to keep all the old documentation (older than latest)

Comment (LucioFranco) [reply generate.sh:82]: @jojolepro its building *JUST* the book. It doesn't hurt us to do it and might be beneficial down the line.

Comment (AnneKitsune) [reply generate.sh:82]: ok :)

**Review by AnneKitsune — APPROVED** [review]

File: generate.sh [hunk 3]
```diff
@@ -33,33 +32,75 @@ cargo install-update --version || cargo install cargo-update;
...
+# Build latest tag
+cd amethyst/master
+LATEST_TAG=$(git describe --abbrev=0 --tags)
+
+echo "Checking out tag $LATEST_TAG to build latest docs and bogok"
+git checkout -q $LATEST_TAG
+
+cargo doc --all;
+
+echo "Compiling master branch book"
```
Comment (dotellie) [inline generate.sh:66]: This message is probably wrong

File: src/sass/doc.scss [hunk 4]
```diff
@@ -47,8 +47,8 @@
 }
 
 #develop {
-    margin-left: 20px;
-    background-color: $develop-color;
+  margin-left: 20px;
```
Comment (dotellie) [inline src/sass/doc.scss:50]: Isn't the indentation here inconsistent?

File: src/templates/doc.html [hunk 5]
```diff
@@ -27,27 +27,27 @@
                 <br>
                 <br>
                 <br>
-                <a href="{{ config.extra.crates_url }}" class="branch-desc master-link">Available from crates.io</a>
+                <a href="{{ config.extra.github_url }}" class="branch-desc master-link">Available from the GitHub repository</a>
             </div>
             <div id="develop">
```
Comment (dotellie) [inline src/templates/doc.html:32]: Can this id be renamed too?

File: src/templates/doc.html [hunk 6]
```diff
@@ -6,8 +6,8 @@
 
 {% block content %}
     <div id="content">
-        <div id="head">Blue pill or red pill?
-            <div id="head-desc">This webside provides documentation for both the master and develop branches. Unless you're comfortable with rapid and constant change, <b>we recommend using the master branch</b>.</div>
+        <div id="head">Documentation
+            <div id="head-desc">This webside provides documentation for the master branch of Amethyst.</div>
```
Comment (dotellie) [inline src/templates/doc.html:10]: Maybe we should keep the old message or at least add a mention of the release docs?

**Review by dotellie — COMMENTED** [review]
You merged while I was reviewing, but as to not have these comments get lost, here they are anyway.

File: generate.sh (hunk 3 above)
Comment (LucioFranco) [reply generate.sh:66]: Yup, missed that, thanks.

File: src/templates/doc.html (hunk 6 above)
Comment (LucioFranco) [reply src/templates/doc.html:10]: Yup, I will update this to be clearer on what this page has. Thanks.

File: src/templates/doc.html (hunk 5 above)
Comment (LucioFranco) [reply src/templates/doc.html:32]: Yeah, I need to go through and rename the develop classes and ids but got kinda lazy.

**Comment by Moxinilian** [comment]
I really would have preferred you did not merge that so quickly, as there’s a bunch of issues I now want to fix.
Why is master the recommended source? It feels like anything but good for a default. The color theming is inappropriate now too, as red is supposed to identify the bleeding edge.
We also absolutely need examples for current releases. Not having it is a no-go for me, as I stated it in other places.
I think the documentation page deserves a complete redesign, because the 2 things architecture no longer is appropriate.

**Comment by LucioFranco** [comment]
@Moxinilian Sorry, I mentioned why I did it quickly on discord. Anyways, we should be recommending either a released version on crates.io or the master branch. The colors we can change, but I don't think its a blocker as of right now. Feel free to fix it or make an issue for it. The examples can be found via this link, https://github.com/amethyst/amethyst/tree/2deccbb8042c1eccdaacc9091ffbf234976f1473 which is the commit for `v0.8.0`. What we need to do is find a way to insert the latest version into gutenberg and use that to build the url to the examples. This is the final link we should use for examples on the website, and we want this to change automatically as we release new tags. https://github.com/amethyst/amethyst/tree/2deccbb8042c1eccdaacc9091ffbf234976f1473/examples

I agree it could use a redesign but right now it does the job, even if it doesn't do it optimally. 😄

**Comment by Moxinilian** [comment]
I got no response when I suggested keeping a “latest-release” branch to make everything simpler regarding both links and release deployment, yet it solves all of the issues at once.

**Comment by LucioFranco** [comment]
@Moxinilian I'm pretty sure, we had agreed in the original issue that we did not need that, and again, a tag is a branch, so there is no need to have a latest that just becomes stale, imo. You should be able to just append a config variable to the gutenberg config with the commit hash and achieve the same result, without needing a latest branch.

**Comment by Moxinilian** [comment]
And as a side effect require either manual maintenance or a hacky generate.sh cluttering the website.

**Comment by LucioFranco** [comment]
@Moxinilian Nope, actually, it is one git command to fetch the latest tag, from that you can get the commit hash. Appened a variable for both the tag and the commit hash. And you can use those to build the URL. It's not hacky and it is actually a lot cleaner and clearer. If you want to hop on discord and continue lets do that. 😄

**Pull request merged**